tests/Authenticator.hs

```diff
@@ -49,17 +55,52 @@ data AuthenticatorSignatureCounter
   = Unsupported
   | Global M.SignatureCounter
   | PerCredential (Map.Map M.CredentialId M.SignatureCounter)
-  deriving (Show)
+
+data SigningBehaviour
+  = GenerateCorrectSignature
+  | GenerateRandomSignature
```

Comment:
I think it might be better to parametrize by malicious behavior. So instead of a random signature, to have a signature that's signed with the wrong (a random) key.

Also I'd name this
```haskell
data SigningBehaviour
  = SigningBehaviourCorrect
  | SigningBehaviourWrongKey
```